lib/routes/profile.js

```diff
@@ -43,27 +56,39 @@ module.exports = {
     }
   },
   handler: function email(req, reply) {
+    const server = req.server;
     var creds = req.auth.credentials;
     if (!hasAllowedScope(creds.scope || [])) {
       return reply(Boom.forbidden());
     }
-    batch(req, {
-      email: '/v1/email',
-      uid: '/v1/uid',
-      avatar: '/v1/avatar',
-      displayName: '/v1/display_name'
-    })
-    .done(function (result) {
-      if (creds.scope.indexOf('openid') !== -1) {
-        result.sub = creds.user;
-      }
-      var rep = reply(result);
-      var etag = computeEtag(result);
-      if (etag) {
-        rep = rep.etag(etag);
+
+    if (!server.methods.batch) {
+      createServerMethod(server);
+    }
+    server.methods.batch(
+      req,
+      {
+        email: '/v1/email',
+        uid: '/v1/uid',
+        avatar: '/v1/avatar',
+        displayName: '/v1/display_name'
+      },
+      (err, result, cached, report) => {
+        if (creds.scope.indexOf('openid') !== -1) {
+          result.sub = creds.user;
+        }
+        if (err) {
+          return reply(err);
+        }
+        var rep = reply(result);
+        var etag = computeEtag(result);
+        if (etag) {
+          rep = rep.etag(etag);
+        }
+        const lastModified = cached ? new Date(cached.stored) : new Date();
```

Comment:
We will need a test for caching and invalidation. Maybe something like this:

1. Fetch resources
2. Fetch again, see if cached
3. Invalidate them
4. Fetch again, make sure they are not cached.

Also a test for the cache default expiry would be good. 


0. Set cache config override to 1 second.
1. Fetch resources
2. Fetch again in 2 seconds, see if **not** cached